### PR TITLE
fix: add required labels to pods

### PR DESCRIPTION
**Pull request by codingben**
**What this PR does / why we need it**:

Add required labels (managed-by, part-of, version, component) to template-validator and vm-console-proxy pods.

Jira-Url: https://issues.redhat.com/browse/CNV-28185

**Release note**:

```release-note
Add required labels to template-validator and vm-console-proxy pods.
```

### CODE REVIEW

File: internal/common/labels.go [hunk 1]
```diff
@@ -47,6 +48,20 @@ func AddAppLabels(requestInstance *ssp.SSP, name string, component AppComponent,
 	return obj
 }
 
+// AddAppLabels to the provided object metadata
+// Name will translate into the AppKubernetesNameLabel
+// Component will translate into the AppKubernetesComponentLabel
+// Instance wide labels will be taken from the request if available
+func AddAppLabelsToMeta(objectMeta metav1.ObjectMeta, requestInstance *ssp.SSP, name string, component AppComponent) metav1.ObjectMeta {
```
Comment (0xFelix) [inline internal/common/labels.go:55]: Be careful with passing around copies of `objectMeta`. The original `AddAppLabels` is working slightly differently, we're the original object is modified. This func takes a copy of `objectMeta` and returns the modified copy instead.

Can you make both `AddAppLabels` and `AddAppLabelsToMeta` use a helper func to do the acutal work between L42-L46 and L56-60, which is identical? (dedup this block)

Comment (codingben) [reply internal/common/labels.go]: Done. Thanks!

**Comment by codingben** [comment]
I've tested the changes in the cluster and it works as expected.

**Comment by jcanocan** [comment]
Nice job @codingben! /lgtm

**Comment by jcanocan** [comment]
Nice job @codingben !
/lgtm

File: internal/common/labels.go [hunk 2]
```diff
@@ -38,15 +39,20 @@ const (
 // Instance wide labels will be taken from the request if available
 func AddAppLabels(requestInstance *ssp.SSP, name string, component AppComponent, obj client.Object) client.Object {
```
Comment (akrejcir) [inline internal/common/labels.go:40]: Can you change the type of the last argument in this function to `metav1.Object` interface? Then you can use it for the deployment templates, and you can remove the new function `AddAppLabelsToMeta()`.

Also the return value of this function is not used anywhere (except tests), so you can remove it if you want.

File: internal/common/labels.go [hunk 3]
```diff
@@ -38,15 +39,20 @@ const (
...
-
+	addCommonLabels(labels, requestInstance, name, component)
 	return obj
 }
 
+// AddAppLabels to the provided object metadata
+// Name will translate into the AppKubernetesNameLabel
+// Component will translate into the AppKubernetesComponentLabel
+// Instance wide labels will be taken from the request if available
+func AddAppLabelsToMeta(objectMeta *metav1.ObjectMeta, requestInstance *ssp.SSP, name string, component AppComponent) *metav1.ObjectMeta {
```
Comment (akrejcir) [inline internal/common/labels.go:50]: This function is not needed, see my above comment.

File: internal/common/labels_test.go [hunk 4]
```diff
@@ -75,4 +75,14 @@ var _ = Describe("AddAppLabels", func() {
 		labels := obj.GetLabels()
 		Expect(labels[AppKubernetesManagedByLabel]).To(Equal("ssp-operator"))
 	})
+
+	It("adds required labels", func() {
```
Comment (akrejcir) [inline internal/common/labels_test.go:79]: After the change, this test can be removed, because it will be covered by other tests.

Comment (codingben) [reply internal/common/labels_test.go]: Done.

File: internal/common/labels.go (hunk 3 above)
Comment (codingben) [reply internal/common/labels.go]: Done.

File: internal/common/labels.go (hunk 2 above)
Comment (codingben) [reply internal/common/labels.go]: Thanks. I wasn't aware of it.

**Comment by codingben** [comment]
@akrejcir @0xFelix @ksimon1 Can you please approve it?

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Comment by codingben** [comment]
Hi @dhiller @brianmcarey, where I can check why kubevirt-bot isn't responding to `/cherry-pick release-v0.18` above?

**Pull request merged**